src/consensus/params.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#include "params.h"
+
+#include "upgrades.h"
+
+namespace Consensus {
+    bool Params::NetworkUpgradeActive(int nHeight, Consensus::UpgradeIndex idx) const {
+        return NetworkUpgradeState(nHeight, *this, idx) == UPGRADE_ACTIVE;
+    }
+
+    int Params::Halvings(int nHeight) const {
+        // zip208
+        // Halving(height) :=
+        // floor((height - SlowStartShift) / PreBlossomHalvingInterval), if not IsBlossomActivated(height)
+        // floor((BlossomActivationHeight - SlowStartShift) / PreBlossomHalvingInterval + (height - BlossomActivationHeight) / PostBlossomHalvingInterval), otherwise
+        if (NetworkUpgradeActive(nHeight, Consensus::UPGRADE_BLOSSOM)) {
+            int blossomActivationHeight = vUpgrades[Consensus::UPGRADE_BLOSSOM].nActivationHeight;
+            // Ideally we would say:
+            // halvings = (blossomActivationHeight - consensusParams.SubsidySlowStartShift()) / consensusParams.nPreBlossomSubsidyHalvingInterval 
+            //     + (nHeight - blossomActivationHeight) / consensusParams.nPostBlossomSubsidyHalvingInterval;
+            // But, (blossomActivationHeight - consensusParams.SubsidySlowStartShift()) / consensusParams.nPreBlossomSubsidyHalvingInterval
+            // would need to be treated as a rational number in order for this to work.
+            // Define scaledHalvings := halvings * consensusParams.nPostBlossomSubsidyHalvingInterval;
+            int scaledHalvings = ((blossomActivationHeight - SubsidySlowStartShift()) * Consensus::BLOSSOM_POW_TARGET_SPACING_RATIO)
```

Comment:
```suggestion
            int64_t scaledHalvings = ((blossomActivationHeight - SubsidySlowStartShift()) * Consensus::BLOSSOM_POW_TARGET_SPACING_RATIO)
```

Reply:
I am unable to commit suggestions, I think since this is not my PR.